onebusaway-gtfs/src/main/java/org/onebusaway/gtfs/model/BookingRule.java

```diff
@@ -0,0 +1,311 @@
+
+   Example: “Ride must be booked 1 day in advance before 5PM” will be encoded as prior_notice_last_day=1.
+
+   Conditionally Required:
+   (The timing must be provided by either a duration or a last time & day)
+   Required for up-to-prior-day booking (booking_type=2).
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_last_day", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeLastDay;
+
+  /**
+   prior_notice_last_time
+   (Time, Conditionally Required) Latest time of day on the last day on which a booking can be made. The timezone used is the one defined by agency.agency_timezone.
+
+   Example: “Ride must be booked 1 day in advance before 5PM” will be encoded as prior_notice_last_time=17:00:00.
+
+   Conditionally Required:
+   Required if prior_notice_last_day is defined.
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_last_time", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeLastTime = MISSING_VALUE;
+
+  /**
+   prior_notice_start_day
+   (Integer, Conditionally Optional) Earliest day on which a booking can be made. Defined as an offset, so the number of service days in advance of the booking.
+
+   Example: “Ride can be booked at the earliest one week in advance at midnight” will be encoded as prior_notice_start_day=7.
+
+   Conditionally Optional:
+   Optional for up-to-prior-day booking (booking_type=2).
+   Optional for up-to-same-day booking (booking_type=1) if prior_notice_duration_max is empty.
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_start_day", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeStartDay;
+
+  /**
+   prior_notice_start_time
+   (Time, Conditionally Required) Earliest time of the earliest day at which a booking can be made. The timezone used is the one defined by agency.agency_timezone.
+
+   Example: “Ride can be booked at the earliest one week in advance at midnight” will be encoded as prior_notice_start_time=00:00:00.
+
+   Conditionally Required:
+   Required if prior_notice_start_day is defined.
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_start_time", optional = true, mapping = EntityFieldMappingFactory.class)
+  private int priorNoticeStartTime = MISSING_VALUE;
+
+  /**
+   prior_notice_service_id
+   (ID from calendar.txt, Optional) When prior_notice_start_day is used, prior_notice_service_id defines a subset of days that count towards the prior_notice_start_day.
+
+   Example: If empty, prior_notice_start_day=2 will be two calendar days in advance. If defined as a service_id containing only business days (weekdays without holidays), prior_notice_start_day=2 will be two business days in advance.
+   */
+  @CsvField(name = "prior_notice_service_id", optional = true, mapping = ???.class)
```

Comment:
Use `AgencyAndId for type` similar to here https://github.com/kyyticom/onebusaway-gtfs-modules/blob/gtfs-flex-2_booking-rules/onebusaway-gtfs/src/main/java/org/onebusaway/gtfs/model/Trip.java#L35